source/dotnet/wholesale-api/Orchestrations.IntegrationTests/Functions/Calculation/CalculationOrchestrationStateTests.cs

```diff
@@ -0,0 +1,205 @@
+            Fixture.TestLogger);
+        isCalculatedState.ActualState.Should().BeOneOf(
+            CalculationOrchestrationState.Calculated,
+            CalculationOrchestrationState.ActorMessagesEnqueuing);
+
+        // => When the calculation result is complete, state should be ActorMessagesEnqueuing
+        // We need to wait for the state change from Calculated to ActorMessagesEnqueuing if it hasn't already
+        // happened in previous step
+        var isActorMessagesEnqueuingState = await dbContext.WaitForCalculationWithState(calculationId.Value, CalculationOrchestrationState.ActorMessagesEnqueuing, Fixture.TestLogger);
+        isActorMessagesEnqueuingState.ActualState.Should().Be(CalculationOrchestrationState.ActorMessagesEnqueuing);
+
+        // => Raise "ActorMessagesEnqueued" event to the orchestrator
+        await Fixture.DurableClient.RaiseEventAsync(
+            orchestrationStatus.InstanceId,
+            MessagesEnqueuedV1.EventName,
+            new MessagesEnqueuedV1
+            {
+                CalculationId = calculationId.ToString(),
+                OrchestrationInstanceId = orchestrationStatus.InstanceId,
+            });
+
+        // => Orchestration is "ActorMessagesEnqueued" or "Completed", state should be ActorMessagesEnqueued or Completed
+        // The state changes from ActorMessagesEnqueued to Completed immediately, so we need to check for both states.
+        var isActorMessagesEnqueuedState = await dbContext.WaitForCalculationWithOneOfStates(
+            calculationId.Value,
+            [CalculationOrchestrationState.ActorMessagesEnqueued, CalculationOrchestrationState.Completed],
+            Fixture.TestLogger);
+        isActorMessagesEnqueuedState.ActualState.Should().BeOneOf(
+            CalculationOrchestrationState.ActorMessagesEnqueued,
+            CalculationOrchestrationState.Completed);
+
+        // => Orchestration is completed, state should be Completed and orchestration output should be success
+        // We need to wait for the orchestration to complete if it hasn't already happened in previous step
+        var completeOrchestrationStatus = await Fixture.DurableClient.WaitForInstanceCompletedAsync(
+            orchestrationStatus.InstanceId,
+            TimeSpan.FromMinutes(3));
+        var isCompletedState = await dbContext.WaitForCalculationWithState(calculationId.Value, CalculationOrchestrationState.Completed, Fixture.TestLogger);
+        isCompletedState.ActualState.Should().Be(CalculationOrchestrationState.Completed);
+        completeOrchestrationStatus.Output.ToObject<string>().Should().Be("Success");
+
+        // => Verify that the expected message was sent on the ServiceBus
+        var verifyServiceBusMessages = await Fixture.ServiceBusListenerMock
+            .When(msg =>
+            {
+                if (msg.Subject != EnergyResultProducedV2.EventName)
```

Comment:
```suggestion
                if (msg.Subject != CalculationCompletedEventeV1SOMETHING.EventName)
```
This should assert for the service bus message which are responsible for starting EDI's orchestration?